src/daft-recordbatch/src/ops/window_states/sum.rs

```diff
@@ -0,0 +1,132 @@
+use std::{
+    marker::PhantomData,
+    ops::{Add, Sub},
+};
+
+use arrow2::bitmap::MutableBitmap;
+use common_error::{DaftError, DaftResult};
+use daft_core::{datatypes::DaftPrimitiveType, prelude::*};
+use num_traits::Zero;
+
+use super::WindowAggStateOps;
+
+pub struct SumWindowState<'a, T>
+where
+    T: DaftPrimitiveType,
+    T::Native: Zero + Add<Output = T::Native> + Sub<Output = T::Native> + Copy,
+{
+    source: DataArray<T>,
+    sum: T::Native,
+    sum_vec: Vec<T::Native>,
+    valid_count: usize,
+    validity: MutableBitmap,
+    _phantom: PhantomData<&'a ()>,
```

Comment:
It looks like we often have this `_phantom` marker here because we end up not storing the series directly. Wonder if it's even worth it to have the specified lifetime tbh. Maybe we should just remove it. For the case of `Series`, cloning it is pretty cheap anyway. Sorry about that